File: src/main/java/live_templates/GetXContext.java
package live_templates;

import com.intellij.codeInsight.template.TemplateContextType;
import com.intellij.psi.PsiFile;
import org.jetbrains.annotations.NotNull;

public class GetXContext extends TemplateContextType {
    protected GetXContext() {
        super("FLUTTER", "Flutter");
    }

    @Override
    public boolean isInContext(@NotNull PsiFile file, int offset) {
        return file.getName().endsWith(".dart");
    }
}

File: src/main/java/live_templates/GetXContext.kt
package live_templates

import com.intellij.codeInsight.template.TemplateContextType
import com.intellij.psi.PsiFile

class GetXContext : TemplateContextType("FLUTTER", "Flutter") {
    override fun isInContext(file: PsiFile, offset: Int): Boolean {
        return file.name.endsWith(".dart")
    }
}